lxd/container_lxc.go

```diff
@@ -3793,6 +3796,15 @@ func (c *containerLXC) setupHostVethDevice(deviceName string, device types.Devic
 		return err
 	}
 
+	// If an IPv6 address has changed, flush all existing IPv6 leases for container.
+	if device["nictype"] == "bridged" && oldDevice["nictype"] == "bridged" && device["ipv6.address"] != oldDevice["ipv6.address"] {
+		networkClearLease(c.Name(), device["parent"], device["hwaddr"], clearLeaseIPv6Only)
+
+		// Queue the interface to be bounched once dnsmasq config has been reloaded to give
+		// container a chance to detect the change and re-apply for the updated leases.
+		c.dnsmasqDownUpIfaces = append(c.dnsmasqDownUpIfaces, device["host_name"])
```

Comment:
@stgraber this may be a bit controversial, so would appreciate your take on it. 

My thinking here was that if we clear the lease for the current assigned IP then there's a chance some other container will be allocated it and cause IP conflicts on the network. Additionally when using IP filtering mode the new host side filter rules are applied for the new IP, which will cause the old IP to stop working.

I thought that by bouncing the host side of the veth pair this would give the DHCP client some ability to detect a change, but without taking down the container side interface that may cause programs listening on that interface to die.

Reply:
I think that's reasonable but that `dnsmasqDownUpIfaces` is a bit confusing to me.

I'm not very fond of storing temporary state on the `containerLXC` struct as one may then think that the struct persists and isn't re-created from scratch for every request. Since your code works for you, I'm assuming that the relevant calls to `setupHostVethDevice` all happen as part of `Update` but it looks a bit confusing to me.

Reply:
@stgraber I've updated to make it clearer that this is only relevant for the Update action. Thanks